model/chunks/chunkFaults.go

```diff
@@ -79,6 +79,35 @@ func NewCFNonMatchingFinalState(expected flow.StateCommitment, computed flow.Sta
 		execResID:  execResID}
 }
 
+// CFInvalidEventsCollection is returned when computed events collection hash is different from the chunk's one
+type CFInvalidEventsCollection struct {
+	expected   flow.Identifier
+	computed   flow.Identifier
+	chunkIndex uint64
+	execResID  flow.Identifier
+}
+
+func NewCFInvalidEventsCollection(expected flow.Identifier, computed flow.Identifier, chInx uint64, execResID flow.Identifier) *CFInvalidEventsCollection {
+	return &CFInvalidEventsCollection{
+		expected:   expected,
+		computed:   computed,
+		chunkIndex: chInx,
+		execResID:  execResID,
+	}
+}
+
+func (c *CFInvalidEventsCollection) ChunkIndex() uint64 {
+	return c.chunkIndex
+}
+
+func (c *CFInvalidEventsCollection) ExecutionResultID() flow.Identifier {
+	return c.execResID
+}
+
+func (c *CFInvalidEventsCollection) String() string {
+	return fmt.Sprintf("events collection hash differs, got %x expected %x", c.computed, c.expected)
```

Comment:
Any reason we do not include the result ID and chunk index?

Reply:
No reason, but it will be helpful, will add